domain/application/state/application.go

```diff
@@ -3645,8 +3645,11 @@ func decodeConstraints(cons applicationConstraints) constraints.Value {
 		if row.ImageID.Valid {
 			res.ImageID = &row.ImageID.String
 		}
-		if row.Space.Valid {
-			spaces.Add(row.Space.String)
+		if row.SpaceName.Valid && row.SpaceExclude.Valid {
+			spaces[row.SpaceName.String] = constraints.SpaceConstraint{
+				SpaceName: row.SpaceName.String,
+				Exclude:   row.SpaceExclude.Bool,
+			}
```

Comment:
What happens if you don't have an exclude? This will only add a space if there is an exclusion, or do we ALWAYS fill in the exclusion when we have a space?

Reply:
My point here is that if someone doesn't insert both at the same time, this breaks.